packages/core/src/tracing/instrumentation/cloud/aws/dynamodb.js

```diff
@@ -0,0 +1,179 @@
+/*
+ * (c) Copyright IBM Corp. 2021
+ * (c) Copyright Instana Inc. and contributors 2021
+ */
+
+'use strict';
+
+const shimmer = require('shimmer');
+const requireHook = require('../../../../util/requireHook');
+const cls = require('../../../cls');
+const { EXIT, isExitSpan } = require('../../../constants');
+const tracingUtil = require('../../../tracingUtil');
+
+const operationsInfo = {
+  createTable: { op: 'create' },
+  // deleteTable: { op: 'delete' },
+  listTables: { op: 'list' },
+  scan: { op: 'scan' },
+  query: { op: 'query' },
+  getItem: { op: 'get' },
+  deleteItem: { op: 'delete' },
+  putItem: { op: 'put' },
+  updateItem: { op: 'update' }
+};
+
+const methodList = Object.keys(operationsInfo);
+
+let isActive = false;
+
+exports.isActive = function() {
+  return isActive;
+};
+
+exports.init = function init() {
+  requireHook.onModuleLoad('aws-sdk', instrumentAWS);
+};
+
+exports.activate = function activate() {
+  isActive = true;
+};
+
+exports.deactivate = function deactivate() {
+  isActive = false;
+};
+
+function instrumentAWS(AWS) {
+  shimmer.wrap(AWS.Service.prototype, 'makeRequest', shimMakeRequest);
+}
+
+function shimMakeRequest(originalMakeRequest) {
+  return function() {
+    if (isActive) {
+      const originalArgs = new Array(arguments.length);
+      for (let i = 0; i < originalArgs.length; i++) {
+        originalArgs[i] = arguments[i];
+      }
+
+      return instrumentedMakeRequest(this, originalMakeRequest, originalArgs);
+    }
+
+    return originalMakeRequest.apply(this, arguments);
+  };
+}
+
+function instrumentedMakeRequest(ctx, originalMakeRequest, originalArgs) {
+  if (!isOperationShimmable(originalArgs[0])) {
+    return originalMakeRequest.apply(ctx, originalArgs);
+  }
+
+  if (cls.tracingSuppressed()) {
```

Comment:
nit: this could be merged with the `if` above. 

```
if (!isOperationShimmable(originalArgs[0]) || cls.tracingSuppressed()) {
    return originalMakeRequest.apply(ctx, originalArgs);
  }
```

But that's mostly a matter of taste I guess.